[RFC 0114] Code of Conduct for NixOS Community (#114)
NixOS/rfcs: rfcs/0114-nixos-code-of-conduct.md

---
feature: NixOS Code of Conduct
start-date: 2021-11-03
author: @jonringer
co-authors:
shepherd-team:
shepherd-leader:
related-issues: #98 #102
---

# Summary
[summary]: #summary

The NixOS Community needs some foundation of shared values which can be used

> to determine appropriate when contributing and interacting with the community.

Reviewer: ```suggestion
to determine appropriate behavior when contributing and interacting with the community.
```

We currently have a [community ethos](https://nixos.org/community/index.html), but

> this only states that we are an inclusive community, but doesn't define

Reviewer: ```suggestion
this only states that we are an inclusive community, and doesn't define
```

Author: I'm terrible at proper english grammar, so I'm probably wrong. But I think `but` is more appropriate as I'm listing what it's doing, `and` to me should be used when doing another thing (not the absence thereof).

inappropriate behavior when interacting with the community. This RFC aims

> to established the shared values, so they can be used to identify behavior

Reviewer 2: ```suggestion
to establish the shared values, so they can be used to identify behavior
```

Author: thanks :)

> to established they shared values, so they can be used to identify behavior

Reviewer 3: ```suggestion
to established the shared values, so they can be used to identify behavior
```

Author: thanks :)

which is disruptive to the community.

This RFC does not intend to define moderation practices. This is intended to only lay
a framework of shared values for the community.

# Motivation
[motivation]: #motivation

> The NixOS community struggles with a clearly defined way in which someone's behavior may

Reviewer: ```suggestion
The NixOS community lacks a clearly defined way in which someone's behavior may
```

be disruptive. A well-defined Code of Conduct gives an explicit
set of expectations for all contributors, which will also make it easier to identify
when behavior becomes disruptive.

# Detailed design
[design]: #detailed-design

These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
which provides a great compromise between explicit behaviors and subjective goals.

> Only the moderation body name has been changed.

Reviewer: ```suggestion
Only minor changes have been made.
```

- We are committed to providing a friendly, safe and welcoming environment for

> all, regardless of level of experience, gender identity and expression,

Reviewer 4: I have a feeling that one of the attributes is not like the others here, and pretending it is will only lead to more events that feel hypocritical to some. Basically, we want the environment to be safe and friendly regardless of level of experience, but at some level it is very much not welcoming in practice, because the environment where all conversations go above your head is not too welcoming. And whatever idealists say, Nix is complicated enough that Nix spaces will be a bad place to learn basic Linux knowledge for most people.

While I agree with @asymmetric about the two additions, I guess having access to something powerful enough to evaluate anything useful is an economic prerequisite to the place succeeding at being welcoming.

Author: > we want the environment to be safe and friendly regardless of level of experience, but at some level it is very much not welcoming in practice, because the environment where all conversations go above your head is not too welcoming.

This sounds more related to having a "noob friendly" area, rather an issue with personal conduct. This is to just to make it clear that people with little experience are also welcome.

> And whatever idealists say, Nix is complicated enough that Nix spaces will be a bad place to learn basic Linux knowledge for most people.

I agree, but I'm not sure what the correlation is with this CoC. Learning materials has always been a pain point in nix's past.

Reviewer 4: I have seen bans that to me looked like bans for being too clueless… Our noobs do have quite a bit background, and learn with reasonable speed or go away frustrated. And the learning materials we are hoping for — they would still start rather high in terms of expected backgrounds (which I find reasonable).

Maybe even I would qualify «all» with «all willing to learn»… It's true that newbies who are below our level but ask how to learn have a reasonable chance to get a recommendation for path to filling in missing background (probably somewhere outside) — and when I say chance, I mostly mean that recommending learning paths is hard, sometimes it doesn't happen or a recommendation is suboptimal.

Author: If someone is being target just because they are new, then obviously that's an issue. But I think the current situation would be improved by just having better on-boarding documentation. Nix pills is a great second resource, but it does a very deep dive into how nix works, but doesn't really help with most nix use cases.

My personal efforts with @nrdxp are to create the "nix book" https://book.divnix.com/ . However, it keeps getting delayed because of other nix related events, like this.

Reviewer 4: Well, they are banned because of their level of cluelessness is declared disruptive (with significant reasons for that declaration, sure, whether sufficient or not). 

Maybe I want to qualify this in some way because the other listed attributes should not even affect the substantial conversation except the relatively special cases («as a colourblind contributor X finds the choices in the markup frustrating» or maybe the converse «you have 100% eyesight sharpness, so I am not surprised you find this font size optimal, but not everyone is like that»), and offtopic conversations also only in similar circumstances. Being inexperienced puts some expectations on the person about learning and maybe even not jumping ahead too much.

Author: > Well, they are banned because of their level of cluelessness is declared disruptive (with significant reasons for that declaration, sure, whether sufficient or not).

I don't have context on this, cc @grahamc for thoughts.

Reviewer 5: A community made more inclusive by…
At that, why is that this one immediately singles people out for their immutable characteristics, rather than include them?

sexual orientation, disability, personal appearance, body size, race, ethnicity, age,

> religion, nationality, or other similar characteristics.

Reviewer: ```suggestion
religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.
```

These are both from the citizen CoC and I feel they were missing here.

Author: Sure, these seem reasonable, and shouldn't come into play within the community.

- Please avoid using overtly sexual aliases or other nicknames that might
detract from a friendly, safe and welcoming environment for all.
- Please be kind and courteous. There’s no need to be mean or rude.
- Respect that people have differences of opinion and that every design or

> implementation choice carries a trade-off and numerous costs. There is seldom a right answer.

Reviewer 4: ```suggestion
implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
```

Reviewer 4: Also, I would maybe put this bullet point higher. This point might be a contributory factor to most GitHub bans…

- Please keep unstructured critique to a minimum. If you have solid ideas
you want to experiment with, make a fork and see how it works.
- We will exclude you from interaction if you insult, demean or harass anyone.
That is not welcome behavior. We interpret the term “harassment” as including the definition in the
[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);
if you have any lack of clarity about what might be included in that concept,
please read their definition. In particular, we don’t tolerate behavior that excludes

> people in socially marginalized groups.

Reviewer 4: I am kind of worried that «deliberate online following» is included there (although indeed the harassment definition there seems otherwise fine)

Author: What would you like to see done?

Reviewer 4: Hm, do you have an interpretation what is their «following» as separate from «stalking» (and preparation of «doxing», I guess)? I guess saying that's how we interpret it would be more than enough.

Author: At least my personal interpretation is that stalking is when you actively involve oneself with someone else's personal life, where as following may just be limited to public life. Both usually have an unwanted connotation.

Since the bullet point in the Citizen CoC mentions both online and personal. I think there were covering the case in which real life was taken into consideration.

Reviewer 4: Hm maybe mention then, that «we interpret following and stalking as either excessive in-person following/stalking, or actively forcing excessive and unwelcome interaction in online spaces»?

Reviewer 3: I think @jonringer's assessment is correct, that the intended meaning is that the "following" not in the social media sense of subscribing to someone's feed. For reference, the statement under question is:

```
Deliberate intimidation, stalking or following (online or in person).
```
ref: https://github.com/stumpsyn/policies/blob/b8ee50e7a9fc79c39c936830901f8863d493b7cd/citizen_code_of_conduct.md

To take a hypothetical; if someone were to start unwelcome discussions with someone in the community, and then continued to initiate those discussions and message them in every possible online venue, it could be considered to be the equivalent of in-person __following__ someone around the entire day. Even if they kept their distance, the persistent intimidation rises to the level of deliberateness that would make the overall behavior unacceptable. It is not a single action of being "followed" or subscribed to on a single social media platform. This matches the common legal definition of stalking that has an element of repetition or longevity.

Reviewer 4: I agree that the intention is as you say; it's just that someone persistently following someone keeping 10 meters distance is probably real-life following; and I basically want to have a pre-agreed opinon on when/whether «constantly reading someone's `with_replies` and then quoting their deep-nested technically public comments» becomes creepy. (Assume that quoting is relevant, acknowledges context and doesn't try to twist their words, bad-faith quoting is bad on its own)

Reviewer 6: How do we define "socially marginalized"? It seems to me there are exponetial ways to exploit and/or distort this concept. If we squint hard enough, we could probably all find at least one, or more likely, several dimensions in our lives in which we could classify ourselves as genuinely "on the margin" in some respect or another.

This means that the benefit goes to the people or groups who spend the most amount of time pondering all the ways in which they are marginalized. 

I just don't understand why the MLK style of reasoning has fallen so dismally out of style lately? What's wrong with, "in particular, we don't tolerate behavior that excludes people on any basis other than the content of their character"?

If we look at current vs past political history for example, the former seems to lead to mass rioting, and a political devide unlike anything I've ever seen, while the latter leads to truly peaceful protest and genuine social reform. Because of that, I find this entire mode of argument highly suspect, to say the least.

We have good examples of how to enact social change, and everyone from MLK to Ghandi all agreed that treating _all people equally_ is absolutely essential to that process. This new style of trying to split everyone into N possible categories is both unsustainable and disingenuous. It leads to conflict, and I dunno how long its proponants can maintain that it does anything _but_ that as the world, and especially the US, become more and more politically divided the more this rhetoric gets pushed around.

All this is to say that we should probably stick with the tried and true method that has actually been _shown to bear fruit_ with respect to bringing people of disparate backgrounds together for a common purpose, and not this new radical philosophy whose only fruit so far is hatred and derision.

Reviewer 7: @nrdxp: I believe it is not necessary to spell out in highly precise detail what socially marginalized means and I believe if we try to do so this thread will derail into an unproductive political science debate.

The purpose of a Code of Conduct is not to be an machine-checked specification that a moderator can robotically execute upon like a smart contract.  A Code of Conduct is a broad statement of principles that can set moderators in the right direction and moderators are generally given discretion when interpreting the Code of Conduct in order to hew to its spirit as closely as possible (just like judges are given discretion in how to interpret the law that best matches the original intent).

If a rogue moderator attempts to subvert the original spirit of the Code of Conduct there are checks upon their behavior: the Code of Conduct can be amended to clarify points of contention, the moderator can be expelled, or in extreme cases the moderator's decisions can be ignored by the rest of the community if they are widely regarded as illegitimate.

Reviewer 4: > Code of Conduct is a broad statement of principles that can set moderators in the right direction

With that goal we can drop the idea of Code of Conduct and use the Foundation Statement.

> If a rogue moderator attempts to subvert the original spirit of the Code of Conduct there are checks

There is no «rogue moderator subverting CoC». There are «contributors that consider a recent moderation action to be a subversion of the project». As long as this set is very small or very large, it is more or less clear what to do, although of course the latter case is annoying. The question is whether some code of conduct can reduce the risk of a rougly even split in some way.

This might not require defenses against novel future subversions of the notions, but maybe it's better to make sure that the notions used do not come pre-subverted in the sense of having an interpretation split from the very beginning.

Reviewer 6: @Gabriel439, you may be right, but the point was mainly offering a very clear alternative to this language which I think many of us could agree on, which offers better precision and less ambiguity in roughly the same amount of words. The point wasn't that we should try to define any particular language, rather to draw attention to the fact that ambiguous language is utterly unhelpful for policy documents at best, and a deliberate mechanism for abuse at worst.
 
I don't agree that it is entirely off topic to reference past precedence and current events, at least here in a policy oriented RFC. It is, by definition, a political discussion after all. I'm sure we could all agree that language is a very powerful tool. It can be quite useful to see how a certain type of rhetoric has played out in the past, or how it is playing out in the present.

> contributors that consider a recent moderation action to be a subversion of the project

I can only speak for myself here, but I didn't necessarily feel any particular moderating action was necessarily subversive to the entire project. More that bad leadership _can_ quite easily become an endemic problem which eventually _will_ become subversive to the project. I don't have any particular gripes with any individual in this community on a personal level, and honestly quite grateful for everyone in this community.

This is just my two cents, but I feel good leaders should do their upmost to check their own political bias at the door when making moderating decisions. Focusing solely on the content of one's behavior rather than what particular groups they may or may not affiliate themselves with is a good way to start IMHO. That said, I do have a great respect both for the tremendous work our leadership has already done, and the incredible burden that acting in a leadership role can bring.

My point isn't to condemn any particular individual for such and such decision. Given the incredible fallibility of being human, I am more than willing to forgive. The point is more that we should have critical discussion, at least from time to time, about actions that leaders may take that could be considered contraversial. After all, how could I expect a person in a position of authority to correct if I've not the courage to confront them on something I feel they may have acted mistakenly.

It's very difficult ground to tread, which I'm sure is no small part of the reason many are so hesitant to tread it. But I'm of the opinion that tackling at a problem at it's source is often the best course of action, and I feel a lot of the discussion so far has been more scraping the surface of the underlying tensions.

We don't all have to air our dirty laundry, but we should be able to have a simple discussion about which language could be considered helpful to our aim here, and which could be considered unhelpful or even outright harmful in the worst case.

> the notions used do not come pre-subverted in the sense of having an interpretation split from the very beginning

If I'm not misunderstanding what you are getting at here, I think preconceived notions are just a fact of life on all sides. A big part of the reason why discussions like this can be helpful is because it allows us to challenge those notions in a structured, and hopefully, productive manner. Imagining that some people come to the table with upfront biases and others do not is just untrue. Nobody is saying that we all have to agree on everything, just that a lot of what we disagree on may be utterly irrelevant to the project, and so is best kept out of day to day affairs.

That said, people are people, and if we don't have and revisit these topics every so often, things have a tendency of boiling over.

Reviewer 4: > I can only speak for myself here, but I didn't necessarily feel any particular moderating action was necessarily subversive to the entire project.

I do not disagree with you! The question is how we make sure that the disagreement on this question stays low (I think we are beyond the scale where zero is a realistic goal…)

> I think preconceived notions are just a fact of life on all sides

Well yes, but just like with things that get people sick, there are questions of degree. At some point it might be worth addressing the questions like yours by trying to drop one level of abstraction or drop the notion completely. Hopefully we can have a text that is no more ambiguous than it has to be.

> people in socially marginalized groups.

Reviewer 4: Separately, if we do «in particular» here, I would say that currently the most relevant thing moderated as harrassment is calling a person in a way they clearly asked not to be called, possibly save for polite references to the reasonably recent and currently relevant contributions/interactions.

Author: I'm not sure what you're asking from this RFC.

In the case of sandro asking not to be called SS2K (because SS), I think that's valid. But this would fall under harassment if that's the case.

Reviewer 4: I want to make things that are implied-bad and have lead to multiple bans to be as prominently discouraged as reasonable. And as far as I remember multiple people have been banned for deliberately and explicitly insisting on using some pronouns in reference to specific people in the community after being asked not to use those.

So I would suggest adding «Neither do we tolerate calling people how they asked not to be called.» and then I guess someone would suggest a better wording.

I agree that Sandro's request was also perfectly reasonable, of course. Fortunately, there things went more amicably.

Author: I think that should be left up to the moderation to arbitrate those incidents.

I think we are getting CoC and enforcement of CoC blurred here. If someone is specifically using the wrong pronoun to harass someone, then that's harassment; and the moderation team should be able to enforce it.

Reviewer 4: «to harass» is an intent, and the logic of those who insist on some kind of essentialism implying the use of the rejected pronouns is alien enough that reaching understanding of intent in any direction is hard… I have seen people go overboard in the other direction, too (explicitly claiming that deliberate avoidance of _all_ pronoun use in some space is harassment via non-recognition of pronoun preference). 

That's why I think calling out a modal example of what actually is not to do (however violations are handled, that's moderation discretion up to a point) could put the intent into a more inter-subjective light.

Author: > «to harass» is an intent, and the logic of those who insist on some kind of essentialism implying the use of the rejected pronouns is alien enough to those who request specific pronouns that reaching understanding of intent in any direction is hard…

genesis (IRC) was banned because he was a non-native speaker and struggled with the use of "they" to refer to a single individual. He was using the [gender neutrl he/him](https://en.wikipedia.org/wiki/Gender_neutrality_in_languages_with_gendered_third-person_pronouns#Generic_he_and_she) instead of the preferred they/them. [The cognitive dissonance](https://logs.nix.samueldr.com/nixos/2020-03-14#3177147), and [subsequent "temp" ban](https://logs.nix.samueldr.com/nixos-dev/2020-03-14#3177453). Difficulty in understanding a world view and language diction which differs from the one you are familiar should not be grounds for removal. For more context, in french "vous" is sometimes used to refer to God. "vous" is also used to refer to [one or more people](https://ai.glossika.com/blog/when-to-use-tu-and-when-to-use-vous) and is most similar to the english "they". An individual wanting to be called "they" could be seen as wanting to be called "God". (I'm intimate with the french language, please correct me if I'm wrong. But it seems to be that genesis struggled with the usage of "they", but didn't mean to cause insult to another contributor).

This is just one example where making pronouns a priority can exclude people that aren't intimate with 21st century American-English and its nuances.

>  I have seen people go overboard in the other direction, too (explicitly claiming that deliberate avoidance of all pronoun use in some space is harassment via non-recognition of pronoun preference).

I'm guilty of this as well. I interact with a lot of strangers on the internet. Unless their pronouns are part of their username, I'm not going to remember. So avoiding pronouns is preferred over incorrect usage.

In an online setting, especially Nixpkgs, I don't think it's reasonable to enforce strangers to be aware of other stranger's pronouns (unless explicit e.g. after the username). If the usage is meant to cause harm, then that falls under the other statutes.

> That's why I think calling out a modal example of what actually is not to do (however violations are handled, that's moderation discretion up to a point) could put the intent into a more inter-subjective light.

Yes, but it's a very specific use case. And the actual disruptive behavior is covered by other statues.

Reviewer 4: Yeah, I also reduce the amount of personal pronouns I use.

About genesis I would say they made some further comments on the topic (on IRC), with the tone that might have affected Graham's decisions. And sure there are pronoun choices that many will not want to follow even if the choice is there in the display name (or even couldn't? what if diacritics is relevant?), which is fine. 

I think the current wording of the guidelines does not _exclude_ the interpretation that writing «calls himself ‘they’» is already outright bannable harassment (this is stronger than the position actually applied in the moderation decision in question), and it also does not clearly _imply_ that weird requests about pronouns should be treated with complete avoidance of pronouns in favour of nick autocomplete (as long as you honestly remember). 

As this is literally an issue where GitHub moderators and Discourse moderators have disagreed in the past, I find it useful to narrow the range of interpretations at least a bit.

Reviewer 8: I think we should remember that the entire pronouns topic is still only really common in the left bubble. Someone not familiar with that will most likely be really confused and do some mistakes even if they don't intent to hurt anyone. I think that is really important to remember. Also NixOS and nixpkgs are first a distro and a package collection, not a social movement and we should choose our fights and not fight every front with radical ideas.

Reviewer 7: @SuperSandro2000: What is the radical idea that you're referring to?

Reviewer 9: Honestly, as a relatively new NixOS/nixpkgs contributor, this is really disheartening to see.

Personally speaking, seeing a prominent contributor express the opinion that using someone's pronouns is a "radical" idea that is "only really common in the left bubble" instead of _basic decency_ is incredibly demoralizing. And the fact that (judging by the reactions) both of the current release managers seem to agree with this sentiment almost makes me want to avoid the project entirely.

I strongly believe that there is no apolitical software. NixOS and nixpkgs are more than a distro and a package collection—they are part of a _community_ of people who contribute to and use them. And this community is made up of real people with real lives, not abstract entities.

Unfortunately, there are no neutral positions here; one either tolerates intolerance in their community or they don't. And your comment seems to imply the former—that, when forced to choose, you would rather have people who are intolerant of others in the community than those they will inevitably push out.

I know my opinion here carries relatively little weight, I just wanted to share my own perspective. I know I can't force you to reconsider, however I would ask that if this choice is made that it be clearly stated in the code of conduct so that new members of the community can be aware of it.

Reviewer 10: unfamiliar/confused != intolerance

Reviewer 9: The comment I am responding to seems to be a statement on "the entire pronouns topic". It is my understanding that this includes people using the wrong pronouns in _all_ situations, including intentionally. It is possible that I misunderstood, however even then I would object to the argument that NixOS/nixpkgs is not a "social movement" being used to justify community moderation decisions, which are inherently social.

Reviewer 10: Moderation of social activities could have a goal of social harmony without having a goal of social change. The more modest goal of social harmony might still require social change on the part of some community members, but it would be a lesser degree.

Reviewer 9: If I understand that correctly then I think I agree. I guess the question then is is who is responsible for changing? Personally, in a situation where someone is being referred to in a way they dislike, I can see two possibilities: either they can decide they are willing to put up with being referred to that way, or the other person can change how they are referring to them. I strongly believe that the latter is the best course of action.

More generally, I think I should probably clarify my stance on this. I would certainly hope that we are not punishing people for honest mistakes, especially since some things (like singular they) can be confusing to non-native speakers. Nor do I expect everyone to use singular they for people they don't know, as nice as that would be. My main concern is people who continue to _intentionally_ refer to someone a certain way after being asked not to. That, from my perspective, is clearly harassment.

The comment I initially replied to seemed to be taking a stance on using others' pronouns as a whole, which is why I responded strongly. Again, it is possible I misunderstood. I certainly hope I did, as that would mean that there is no disagreement here, but I don't know if that is the case right now.

Reviewer 3: @InternetUnexplorer I interpret the original comment as highlighting that NixOS is global, linguistically and culturally diverse, and that we should be inclusive to the best extent possible. I believe we agree about the overall end-goal, to not allow harassing behavior. Perhaps changing the topic and setting slightly would provide a way to analyze a hypothetical. It is a common bullying tactic to use unkind nicknames or diminutives as a way to cause repeated verbal harm. At some point, repeated incorrect usage of someone's name in a particular venue - in spite of being corrected - rises to the level of harassment. At the same time, it would also be harassment to deliberately follow (see discussion above about intimidation, stalking, following) someone to every online venue to correct their language, depriving them of the equivalent of personal space. The overall goal is to reduce harassment; hence the reference to another document which attempts to provide additional commentary.

The point of contention, and it seems this is the more fundamental question, is about neutral positions. Are there neutral positions we take? If the focus is to remain on the distro, package collection, and related technologies, how do we handle the case where some view a position as a binary with no middle or neutral ground?

Reviewer 9: I agree that repeatedly using the wrong name for someone would be similar to using the wrong pronouns, even if the two situations are not quite the same. And I think that the point at which it becomes harassment would be very early on; once someone has been explicitly asked not to refer to someone a certain way, to continue to do so seems to me like a deliberate choice. Of course, mistakes happen now and then, but in my personal experience the difference between making an effort and not caring at all is large enough that it's pretty easy to tell which is which.

As for neutral positions, I really do believe that a decision like this is binary. Either one stands up against this behavior when they see it happening or they don't. To stay out of the situation is a decision to allow that behavior; there isn't really a way they can take a "neutral" position of both allowing it and not allowing it. Of course, where exactly the line is that determines whether something is harassment is much more difficult to decide; I didn't mean to sound like I was implying that decision was binary.

The comment I initially responded to seemed, to me, to be taking the position that there was a neutral option in the matter, and that the neutral option was to allow it/the non-neutral or "radical" position was to not allow it.

Specifically, it said that "the entire pronouns topic is only really common in the left bubble", which seems to me to be saying that a) the "center" or "neutral" position would be to not discuss the entire pronouns topic at all, and b) that those who do are in a bubble, i.e. that their understanding of reality is warped or not indicative of the majority. That, specifically, is what I was referring to when I said I found it disheartening.

Reviewer 11: I think there is value in explicitly stating that we consider it unacceptable to repeatedly refer to someone in a way they have asked not to be referred to.

> We will exclude you from interaction if you insult, demean or harass anyone.

I think this covers both cases of insulting or demeaning nicknames as well as repeatedly using the wrong pronouns to refer to someone.

> Sexist, racist, homophobic, transphobic, ableist or otherwise discriminatory jokes and language.

From the linked Citizen Code of Conduct, I think this bullet would also cover misgendering someone by using the wrong pronouns repeatedly.

In both of the above cases, however, I don't know if my interpretation is universal. The question of whether we need to use people's pronouns seems to keep coming up, so my opinion is that we should have a clear stance on the subject with our reasoning rather than continuously re-litigate it. I hope it would also make it easier to explain to people with less experience in English why we try to respect people's preferred pronouns even if it may not be what they are used to.

> So I would suggest adding «Neither do we tolerate calling people how they asked not to be called.» and then I guess someone would suggest a better wording.

I would suggest "It is unacceptable to repeatedly refer to someone in a way that they have asked not to be referred to. We strive to use people's chosen names and pronouns to show them that they are respected and welcome."

Hopefully this communicates that mistakes happen, and we understand using the wrong pronoun or nickname by accident is not serious. It may be better to replace "repeatedly" with "intentionally", but I don't want to litigate anyone's intentions and it would be the only place in the document to focus on intention (which we can't see) instead of action. I hope this doesn't make it seem like we expect everyone to magically know everyone else's pronouns at all times, the second sentence is meant to be in the same spirit as other parts of the document that provide reasoning such as "There is seldom a single right answer" or that a rule exists to ensure a "welcoming environment for all."

Reviewer 4: > I would suggest "It is unacceptable to repeatedly refer to someone in a way that they have asked not to be referred to. We strive to use people's chosen names and pronouns to show them that they are respected and welcome."

I would slightly weaken the latter part.  «We strive to follow people's preference on names and pronouns as a part of basic expectations of politeness.» On the one hand, even if you as a moderator ban a person, you cannot convincingly claim you still welcome the troll at that specific moment, but should not intentionally misgender the banned person in the process. On the other hand, everyone chooses which option consistent with the preferences to pick. I will not make an effort to look up and use pronouns when I am in conversation with fellow contributors I barely know, I will probably just refer to everyone (including those in the conversation I do know, or those who somehow display the pronouns) by the usernames they themselves have chosen on the platform (or by displaynames, depending what the platform shows more prominently).

Reviewer 12: > genesis (IRC) was banned because he was a non-native speaker and struggled with the use of "they" to refer to a single individual. He was using the [gender neutrl he/him](https://en.wikipedia.org/wiki/Gender_neutrality_in_languages_with_gendered_third-person_pronouns#Generic_he_and_she) instead of the preferred they/them. [The cognitive dissonance](https://logs.nix.samueldr.com/nixos/2020-03-14#3177147), and [subsequent "temp" ban](https://logs.nix.samueldr.com/nixos-dev/2020-03-14#3177453).

I don't think this is true.

The message directly after the one you link is genesis saying "my self i'm a blue wolf with 2 pink nose and one green big finger."  My interpretation of this is that it's a transphobic comment attempting to ridicule WORLDofPEACE's non-binary identity.  It doesn't make sense that somebody who's just confused about how to speak English but otherwise engaging in good faith would come out with a comment like this, because it's not about English, it's about identity, specifically trans identity.

Of course it doesn't make sense to ban somebody for making language mistakes, but that's not what happened here.  I can think of at least one case where a non-native English speaker was told that "he" was not an appropriate way to refer to WORLDofPEACE, and hadn't come across the concept of singular they before.  In that case, they made it clear they didn't mean any disrespect, and there was never a problem with that person again.

Reviewer 2: This is closed and shut already:

> 21:28 <genesis> yes adisbladis but i'm not the one to convinct here
> 21:28 <genesis> but the "THEIR"

Calling WoP "the 'THEIR'" this way is proof of malice you directly linked to. Especially if you know french and know the badly translated turn of phrase used here. "the 'THEIR'" here was clearly used maliciously to refer not to WoP the person, but to WoP "the pronoun using [person]". This, to me, clearly shows they understood.

Note: "convinct" here could be mistaken for "convict". In this exchange it's "convince" (convaincre). In reference to WoP's  reviews on the PR.

Reviewer 2: (Now reading the full comment partially quoted, for due diligence.)

While I totally agree, and expect any "enforcement" to consider the background of the individual, this case wasn't one of those. It would be hypocritical to not consider the background of the individual when considering issues, as *things are different in different locales*.

Now, about the content of the discussion:

> For more context, in french "vous" is sometimes used to refer to God.

I don't think this is right. Though maybe it's a way older deprecated usage that's been continuously taught in french as a second language, or something that didn't stick when the language travelled to the Americas. With that said I couldn't find confirmation about "vous" being used to refer to god anywhere in my usual language resources.

Though, what I had in mind was that "il" (he) sometimes is used to refer to god, but as a narrative construct. Not as a form of respect. But I couldn't confirm either as I wasn't raised in that kind of background. (Or maybe it's "ils", but that wouldn't make sense in a monotheistic religion.)

> "vous" is also used to refer to one or more people and is most similar to the english "they".

Not exactly. It's one person *in polite context*. It's "tu" (you), but polite. It's not even the most similar to the english "they", it's most similar to "you", where in english "you" is both singular and plural all the time.

> An individual wanting to be called "they" could be seen as wanting to be called "God". (I'm not intimate with the french language, please correct me if I'm wrong. But it seems to be that genesis struggled with the linguistic usage of "they" to mean an individual, so the individual must therefore be more than just a single individual (e.g. collective), but didn't mean to intend to cause insult to another contributor).

As I said, I can't disprove that it would be a reference to god in some context, but I wasn't able to find confirmation in my usual language resources.

The more correct (and charitable) interpretation is *they* will be interpreted as "ils/elles", which is always plural, and gendered still. This can make internalizing the differences harder.

This particular case is harder to discern as it's both a gender neutral pronoun, and a plural pronoun use. Most discussions around "they", and gender neutrality has been around "singular people".

So anything about a language you don't master should be taken with a pound of salt. As I do in every languages.

As a final note, while yes initially the confusion has to be tolerated from native french speaking people, **continuous** use of the wrong pronoun cannot be, and then **ridiculing** cannot be tolerated.

Reviewer 8: > Personally speaking, seeing a prominent contributor express the opinion that using someone's pronouns is a "radical" idea that is "only really common in the left bubble" instead of basic decency is incredibly demoralizing.

That's not what I wanted to convey. I wanted to keep my comment short which afterwards left it open for to much interpretation. Sorry for that. Singular they and to more extend neo pronouns is something which I personally don't feel comfortable with expecting from everyone to be fluent in. To make it clear: It is not acceptable to use someones "born with" pronoun that is otherwise completely not fitting.

> As for neutral positions, I really do believe that a decision like this is binary. Either one stands up against this behavior when they see it happening or they don't.

If you are not familiar enough with a topic, don't have enough information or didn't think much about it yet there is a third choice: I don't have an opinion about that, I am not decided yet. We need to remember that not everyone is highly informed about the social issues of our time which is not great but okay.

> especially since some things (like singular they) can be confusing to non-native speakers

That is my point. We have a really high variance of English skills in our community and I don't think we consider this enough. I would rate my English understanding quite high but my reproduction skills and especially my grammar is not that high and more average. My German grammar is also not that great but for example: I can write a gender neutral texts in a good speed but I can't do that in English. My skills are not good enough to write a longer text in a gender neutral way and correct grammar and we are putting really high pressure on people which do not have such a high  skill level.

If you intentionally miss gender someone with he/she/etc in your native language you are quite accountable for that but I don't believe we should put that on a level with people for which English is the second/third/etc language and they are still learning it.

> My main concern is people who continue to intentionally refer to someone a certain way after being asked not to. That, from my perspective, is clearly harassment.

That is not what I try to discuss. Of course that is not okay, we are no longer in a school yard but not every occurrence of this is immediately harassment.

> Specifically, it said that "the entire pronouns topic is only really common in the left bubble", which seems to me to be saying that a) the "center" or "neutral" position would be to not discuss the entire pronouns topic at all and b) that those who do are in a bubble, i.e. that their understanding of reality is warped or not indicative of the majority. That, specifically, is what I was referring to when I said I found it disheartening.

That was not phrased correctly from me. There are basically three stances on this:
- The (far) right which talk the entire day about pronouns and gendering destroying the nature of the language and it will be end of the country and civilization as we know it.
- The neutral ground which don't discuss it at all.
- The left which accepted pronouns and gendering and don't talk to much about it.

> I hope this doesn't make it seem like we expect everyone to magically know everyone else's pronouns at all times

I think this can be interpreted like this which is a problem. Also checking someones profile description before answering a question feels like stalking.

Siamo tutti antifascisti ✊🏼 🏴

Reviewer 9: It seems I misunderstood your comment, then. Apologies and thank you for clarifying!

Reviewer 8: Not a problem. That's why we are talking about and that's why I prefer conversation in real time and person because there you can briefly interrupt someone if there is a clear misunderstanding of what you intended to say.

- Private harassment is also unacceptable. No matter who you are, if you feel
you have been or are being harassed or made uncomfortable by a community member,
please contact one of the channel ops or any of the NixOS moderation team immediately.
Whether you’re a regular contributor or a newcomer, we care about making this community
a safe place for you and we’ve got your back.
- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.

# Examples and Interactions
[examples-and-interactions]: #examples-and-interactions

> There have been a few spectacular failures of communication and good faith within the community.

Reviewer 7: I think the proposed statutes are fine, but I'm not convinced that that they alone would have resolved the three threads referenced here.  If you're going to reference these threads to motivate the statutes then you'd need to elaborate on how a hypothetical moderator would have applied the statutes to improve the outcome of those threads.

As a concrete example, what should a moderator have done differently within the block evasion thread compared to what actually transpired?

Reviewer 4: I guess the _hope_ is that it would take a reasonably short time to reach the conclusion that the proposed  guidelines imply «please don't do it», and there would be something that can imply one way or another in the first place.

Reviewer 13: It might not be useful to use the block evasion thread as an example, because many posts in it were split into a private conversation.

Reviewer 7: Then perhaps we should remove it as an example from this section if it is not an instructive example?

I would still like to see at least one illustration of how one of these motivating examples would have been moderated differently if these guidelines had been place

Author: > need to elaborate on how a hypothetical moderator would have applied the statutes to improve the outcome of those threads.

That's specifically out-of-scope for the RFC. I'm just defining the criteria in which people should act, not how it should be enforced. That's far too much scope for a single RFC.

When a moderator does decide to step in, this CoC is meant to be the basis in which actions are considered.

Reviewer 7: Then perhaps remove the examples?  The only reason I suggest this is that the examples suggestion seems to suggest that this RFC is intended to address those conflicts, but it seems like that connection has yet to be made

Author: Where does it suggest it? I need to change it if it does.

> This RFC does not intend to define moderation practices. This is intended to only lay a framework of shared values for the community.

Reviewer 7: The RFC says:

> In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
better arbitrated as to have a more satisfactory resolution before escalation. This
Code of Conduct will better equip the moderation process by providing clear expectations
for behavior within the community.

… and so I'm asking if this claim could be substantiated by illustrating how it would better equip the moderation process.  If substantiating this claim is out of scope of the RFC then perhaps remove the claim?

Author: I added the PR which spawned the discourse discussion.

Author: > … and so I'm asking if this claim could be substantiated by illustrating how it would better equip the moderation process. If substantiating this claim is out of scope of the RFC then perhaps remove the claim?

Because the current process is essentially graham getting bat phoned that "something is wrong", after the fact. If there was a moderation body then this CoC would be used to identify what behavior was disruptive.

In the case of [this incident](https://github.com/NixOS/nixpkgs/pull/120729#discussion_r621885348), a community member took it upon themselves to act as an arbitrator (which is an indicator that there is a need for a moderation body which is easy to contact). Having an CoC would at least be a guiding reference where the behaviors of both individuals could be judged in a non-partial manner; instead of using language like "be nicer", the language could have been, "Please be kind and courteous. There’s no need to be mean or rude. It's one of the community's core values".

A more active moderation team (that is also easy to contact a member) is still needed for arbitration or moderation. But that's out-of-scope.

Reviewer 5: Absolutely not. Volunteer interaction is what community is.
Thinking a moderation body serves community interests is as far-fetched as thinking the CoC content is.
Not that it follows that a moderation body would follow it either.

Most notable recent examples include the [block evasion thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).

> Each of these incidents have caused significant discussion on IRC (when it was still official),

Reviewer: ```suggestion
Each of these incidents has caused significant discussion on IRC (when it was still official),
```

Reviewer 14: Have is grammatically correct

Reviewer 15: I think that's a difference between flav(o|ou)rs of English, `has` feels right to me as someone who's learned British English

Reviewer: “Each” is iterating over the elements of the list no? And in every iteration the subject is singular

Author: Ironically, I originally wrote `has`, then my word processor recommended `have`, and I changed it because incidents is a plural word.

Reviewer 15: Never trust the grammar check :D

Reviewer 4: I have found online _some_ style guides that seem to be OK with «each … have», and some people here even think it flows better, so it's not so clear of a case! (But yes, I also find «has» more fitting).

Author: I think it makes two different past tenses, both of which may be correct in this instance.

As long as the meaning is clear, I'm not going to favor one or the other.

matrix, discourse, and would bleed over into github and even non-official platforms.
These discussions are generally very polarizing, and causes an enormous amount
of emotional and mental stress to those involved.

> These incidents are also very embarressing for the greater nix community,

Reviewer 16: Embarrassing

Author: ooops

Author: resolved in another edit

> These incidents are also very embarressing for the greater nix community,

Reviewer: ```suggestion
These incidents are also very embarrassing for the greater nix community,
```

and not having a way to identify disruptive before it becomes a heated
issue is detrimental to the health of the community.

In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
better arbitrated as to have a more satisfactory resolution before escalation. This
Code of Conduct will better equip the moderation process by providing clear expectations
for behavior within the community.

# Drawbacks
[drawbacks]: #drawbacks

- There are more "rules" for the community to follow. However, these statutes are intended to

> still allow individuals to participate within the community in a healthy way for all.

Reviewer 4: I guess as a counterpoint you could also claim that this document mostly aims to codify things that are already expected, and in particular flagrant violation of most of these guidelines already leads to moderation actions today.

Author: Sure, seems reasonable

Author: done

# Alternatives
[alternatives]: #alternatives

- #98 Also provides an opinionated values and goals for the moderation team. However,
these rules are less well-defined, and allow for broader interpretations
of unacceptable behavior.

- #102 Codifies some of the existing moderation practices, and uses the

> existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.

Reviewer 4: Is #102 actually an alternative? I would say, keep only having community ethos as the guidelines — with or without #102 or any alternative to it. And I think #102 does not discourage defining the expectations in a separate document.

Author: Yes, but it's not specific about behaviors which have been causing issues.

102 isn't mutually exclusive with this PR. It could be viewed as supplemental.

Reviewer 4: Yes, I just think the alternative to mention here is «Continue using just the statement of community ethos with no more detailed guidelines.», not something about #102

Author: Ah, fair.

Author: added

# Unresolved questions
[unresolved]: #unresolved-questions

# Future work
[future]: #future-work

> - Ratification of a "NixOS moderation team" as described in the second to last bullet.

Reviewer: ```suggestion
- Ratification of a "NixOS moderation team" as described in the second to last bullet point.
```

---
feature: NixOS Code of Conduct
start-date: 2021-11-03
author: @jonringer
co-authors:
shepherd-team: @edolstra @zimbatm @winterqt
shepherd-leader:
related-issues: #98 #102
---

# Summary
[summary]: #summary

> The NixOS Community needs some foundation of shared values which can be used

Reviewer 5: _Has_ some shared values. Evidently don't need these.
Ought-smuggling.

to determine appropriate behavior when contributing and interacting with the community.

> We currently have a [community ethos](https://nixos.org/community/index.html), but

Reviewer 5: Well if singling people out for their immutable characteristics didn't work, I don't know what will…
Oh yes, it surely will help in the slightest to do it over and over again for every place and project anyone visits.
This is the opposite of cognitive behavioural therapy, which is the strongest non-psychotropic effect found in clinical psychology to date.

this only states that we are an inclusive community, but doesn't define
inappropriate behavior when interacting with the community. This RFC aims
to establish the shared values, so they can be used to identify behavior
which is disruptive to the community.

This RFC does not intend to define moderation practices. This is intended to only lay
a framework of shared values for the community.

# Motivation
[motivation]: #motivation

The NixOS community lacks a clearly defined way in which someone's behavior may
be disruptive. A well-defined Code of Conduct gives an explicit
set of expectations for all contributors, which will also make it easier to identify
when behavior becomes disruptive.

# Detailed design
[design]: #detailed-design

> These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),

Reviewer 2: I didn't want to comment this. I hoped I wouldn't need to, but here goes.

I think any CoC which lists specific "classes", or behaviours, or anything of the sort is bad. They will always end up being used for unfun rounds of "rules lawyering" from **any** side. This can already be observed in these comments. People are already LARPING "rules lawyering" about the CoC in this thread.

I have my own short list of CoCs I looked at that had better wording. I won't suggest any at this point in the hope that other individual look into other state of the art CoCs and consider then under that lens. I wouldn't want my personal choices to influence their investigation in the subject.

Reviewer 8: I am not sure how this could be improved by being more generic or on a higher level because I am missing an example I could look at.

Maybe I am even totally misunderstanding what not listing any classes or groups of behavior in practice means.

Reviewer 2: (Note: what I'm saying her is not specific to the chosen CoC, and may or may not apply. They are observed problems.)

Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

Conversely, it may even be used as a way to bully "non-protected" "non-minorities", as the phrasing of such CoCs often makes it "rules lawyerable" to be okay to bully and harass anyone that's not listed.

It's hard to provide examples for this. I don't want to LARP-rules-lawyer bad scenarios, since it's not an easy thing for me to do. Additionally there's the risk of making up bad scenarios that will only be used to attack the ridiculousness of it, rather than the actual problem. Providing "real" examples is often unsavoury; I don't like digging up past drama in projects I've never been involved with, especially when I maybe don't know the full situation(?).

I just know that paradoxically, when you include things in a list, you invariably end up excluding.

Reviewer 8: > Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

That could probably be reduced by phrasing those section like ``for example A, B or C`` or like it is done a few lines below ``other similar characteristics``. I think it should be pretty easy to say that X is similar enough to B and be done with it.

If we dig up more such cases which don't really fit we can always add them later. I even think we should revisit this CoC at some point because we can't potentially cover all cases and everything beforehand.

Reviewer 2: We already have people in these threads unhappy with these lists. It's clear enough that lists are not good. And here you are rules-lawyering about it in these replies already :/.

Reviewer 17: > Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.
> 
> Conversely, it may even be used as a way to bully "non-protected" "non-minorities", as the phrasing of such CoCs often makes it "rules lawyerable" to be okay to bully and harass anyone that's not listed.

This is precisely the reason why I'm usually against any code of conducts. Everyone should be safe from harassment, but listing specific groups implicitly protects only them (Some common groups which are not listed in CoCs are political opinions and affiliations, and personal preferences). In addition, what is considered marginalized depends on region and culture. What might be a majority somewhere, could be a marginalized minority elsewhere.

I don't think we need to make a list of things, which are not allowed to be used as a basis of exclusion or harassment. Instead, I think no-one should feel excluded or harassed, and when exclusion or harassment inevitably occurs, mediating between the parties should be preferred to banhammering the guilty party.

Reviewer 18: > Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

Can you list the specific A Bs and Cs that are listed as “protected clases” by this RFC? I fail to see them.

> Everyone should be safe from harassment, but listing specific groups implicitly protects only them

This is just not true. It is a normal practice in law-like texts to list specific examples and then add a “or similar (in spirit)” catch-all at the end to signify that it’s not a closed-off list and up to the specific case (that’s also why you *need* judges to interpret the law and can’t just write a program as some nerds tend to believe).

As far as I can see, the linked rust CoC has the phrasing “[…] or other similar characteristic.” which is exactly such a clause.

Reviewer 8: > LARPing 

Side note: what is LARPing? A quick google only revealed live action role play which seems to be unfitting.

Reviewer 14: TL;DR what samueldr means is that someone was pretending to be a rules lawyer

Reviewer 2: To be more precise, doing "rules lawyering", "role playing", on either imagined cases or theoretical cases.

Reviewer 2: > This is just not true. It is a normal practice in law-like texts to list specific examples and then add a “or similar (in spirit)” catch-all at the end to signify that it’s not a closed-off list and up to the specific case (that’s also why you need judges to interpret the law and can’t just write a program as some nerds tend to believe).

Until they are invariably rules-lawyered either to work their way around consequences, or worse, *advocate harassment was okay*.

> Can you list the specific A Bs and Cs that are listed as “protected clases” by this RFC? I fail to see them.

Ugh... Assuming good faith from you, and not purposefully missing the point: 

> [...] level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.

And as much as I would like to assume we can trust the community to not do rules lawyering and making my point moot, first, it's not necessarily about *existing members*, which is something to remember, but really, it's about the fact that it seems we cannot assume basic conduct from anyone. If we were able to assume good basic conduct from everyone, the whole discussion here would be moot.

So under the assumption that we can't assume basic conduct, we can't have wording that can be abused. It is trivial for non-toxic persons to miss the point. And given your reaction, I assume you're probably non-toxic here!

* * *

The wording of a CoC cannot be done *in good faith* if the target audience is acting *in bad faith*. This might be a more distilled overview of my point of view.

Reviewer 4: > The wording of a CoC cannot be done in good faith if the target audience is acting in bad faith. This might be a more distilled overview of my point of view.

Actually, maybe we should discuss who the target audience is. 

Personaly, I would consider a desirable outcome if (a) people switching between communities with significantly different norms can understand which mix we have or try to have here, and (b) so that a supermajority of contributors can come to an agreement that moderation is consistent with itself and with the written guidelines. Maybe also (c) the people taking some counter-actions, not exclusively moderators, can discuss propriety of those actions among themselves using a common vocabulary and common set of guidelines.

Indeed, (b) and (c) are open to rule-lawyering, but not as much to malicious rule-lawyering.

which provides a great compromise between explicit behaviors and subjective goals.
Only minor changes have been made.

- We are committed to providing a friendly, safe and welcoming environment for
all, regardless of level of experience, gender identity and expression,
sexual orientation, disability, personal appearance, body size, race, ethnicity, age,

> religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.

Reviewer 19: ```suggestion
religion (or absence thereof), socioeconomic status, nationality, or other similar characteristics.
```

"lack" sounds like something's missing, and that wouldn't reflect the views of those concerned.

- Please avoid using overtly sexual aliases or other nicknames that might

> detract from a friendly, safe and welcoming environment for all.

Reviewer 5: No, toxicity is a measure of how deadly something is in a group of 1000.

- Please be kind and courteous. There’s no need to be mean or rude.
- Respect that people have differences of opinion and that every design or
implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
- Please keep unstructured critique to a minimum. If you have solid ideas
you want to experiment with, make a fork and see how it works.

> - We will exclude you from interaction if you insult, demean or harass anyone.

Author: I would assume this would mean any official platform

> - We will exclude you from interaction if you insult, demean or harass anyone.

Reviewer 5: Chilling.

That is not welcome behavior. We interpret the term “harassment” as including the definition in the

> [Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);

Reviewer 20: Please use an exact revision and/or link to archive.org when linkting to documents that could potentially change.

Author: used the lastest commit

if you have any lack of clarity about what might be included in that concept,
please read their definition. In particular, we don’t tolerate behavior that excludes
people in socially marginalized groups.
- Private harassment is also unacceptable. No matter who you are, if you feel
you have been or are being harassed or made uncomfortable by a community member,

> please contact one of the channel ops or any of the NixOS moderation team immediately.

Reviewer 19: ```suggestion
please contact one of the channel ops or any of the moderation team members immediately.
```

Not sure about what channel ops is supposed to mean. Channel operators? I didn't know we even have such a thing. This text should link to those points of contact. I wouldn't know how to find them.

Whether you’re a regular contributor or a newcomer, we care about making this community
a safe place for you and we’ve got your back.
- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.

> 

Reviewer 21: Also add the CODE_OF_CONDUCT.md in the .github repo, so it's visible on all repos.

# Examples and Interactions
[examples-and-interactions]: #examples-and-interactions

There have been a few spectacular failures of communication and good faith within the community.

> Most notable recent examples include [ad-hominem attacks in a PR review](https://github.com/NixOS/nixpkgs/pull/120729#discussion_r621885348)
> and associated [block evasion discourse thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
> heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).

Reviewer 19: ```suggestion
Most notable recent examples include ad-hominem attacks in a PR review
and an associated Discourse thread about block evasion,
and heated discussions in RFC 98 and RFC 111.
```

I would prefer not to link to that evidence, I made the mistake of taking a look and regretted it instantly.

Author: It was recent for some of the items over a year ago, not so recent now. The purpose of the section was to show that there has been issues for years within the nix community. The examples are also cases where I think everyone believed they were doing the "right" thing, and yet there was still fighting. It demonstrates an escalation of community activity because there was little moderation capacity in place.

Each of these incidents have caused significant discussion on IRC (when it was still official),

> matrix, discourse, and would bleed over into github and even non-official platforms.

Reviewer 19: ```suggestion
Matrix, Discourse, and would bleed over into GitHub and even non-official platforms.
```

> These discussions are generally very polarizing, and causes an enormous amount

Reviewer 19: ```suggestion
These discussions are generally very polarizing, and cause an enormous amount
```

of emotional and mental stress to those involved.

> These incidents are also very embarrassing for the greater nix community,

Reviewer 19: ```suggestion
These incidents are also very embarrassing for the entire Nix community,
```

Not sure whether it makes sense to implicitly distinguish sub-sets of the Nix community, especially since they were not even referred to in this text.

> and not having a way to identify disruptive before it becomes a heated

Reviewer 5: With the "disruptive behavior" identified, if not sorted in the official channels, it gets sorted elsewhere. Or via other methods. Short-circuiting that with bans doesn't prevent disruption, it _is_ disruption.
The attempt may be laudable, but the result is embarrassing.
It boggles my mind that with the obvious pushback on these authoritarian ideas alone someone still thought it would be their calling to just fine-tune the anti-community language to where it could only sound rosy enough.
If the community at large (supposedly) can't function, why is it in the power of anyone to instrument erosion of freedom to that effect?

issue is detrimental to the health of the community.

In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
better arbitrated as to have a more satisfactory resolution before escalation. This
Code of Conduct will better equip the moderation process by providing clear expectations
for behavior within the community.

# Drawbacks
[drawbacks]: #drawbacks

- There are potentially more "rules" for the community to follow. However, many of these rules
are enforced already under the current moderation team, but without an explicit definition.

# Alternatives
[alternatives]: #alternatives

- #98 Also provides an opinionated values and goals for the moderation team. However,
these rules are less well-defined, and allow for broader interpretations
of unacceptable behavior.

- #102 Codifies some of the existing moderation practices, and uses the

> existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.

Reviewer 19: These should be linked. And the nixos.org piece should be a permalink as above.

- Use existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.
However, this only includes inclusive statements, and doesn't define unacceptable behavior.

# Unresolved questions
[unresolved]: #unresolved-questions

# Future work
[future]: #future-work

> - Ratification of a "NixOS moderation team" as described in the second to last bullet point.

Reviewer 22: This can be removed, since RFC 102 was merged and we have a moderation team.

Reviewer 19: ```suggestion
```

- Add Code of Conduct to nixos.org, and mention the Code of Conduct on all relevant

> repositories under the NixOS organization.

Reviewer 22: This should probably be in the "detailed design" section.
